sql/sql_optimizer.cc

```diff
@@ -6268,9 +6268,13 @@ static ha_rows get_quick_record_count(THD *thd, JOIN_TAB *tab, ha_rows limit,
     keys_to_use.merge(tab->skip_scan_keys);
     MEM_ROOT temp_mem_root(key_memory_test_quick_select_exec,
                            thd->variables.range_alloc_block_size);
+    table_map const_tables = tab->join()->found_const_table_map;
+    table_map read_tables = tab->join()->is_executed()
+                                ? (tab->prefix_tables() & ~tab->added_tables())
```

Comment:
is the not operator intentional ? `~`table->added_tables() or should it be logical not `!`?

Reply:
Right, indeed tables() returns  a bitmap
`  constexpr const table_map INNER_TABLE_BIT{1ULL << (MAX_TABLES + 0)};
  constexpr const table_map OUTER_REF_TABLE_BIT{1ULL << (MAX_TABLES + 1)};
  constexpr const table_map RAND_TABLE_BIT{1ULL << (MAX_TABLES + 2)};
  constexpr const table_map PSEUDO_TABLE_BITS{
      INNER_TABLE_BIT | OUTER_REF_TABLE_BIT | RAND_TABLE_BIT};
`

Also I see the same logic in iterator initialization in DynamicRangeIterator::Init(). So this looks good.